tiny_dnn/core/framework/tensor.h

```diff
@@ -61,45 +63,173 @@
 
 namespace tiny_dnn {
 
+template<typename U = float_t>
```

Comment:
What about adding template parameter `Allocator`
```
template<
    typename U = float_t,
    typename Allocator = aligned_allocator<U, 64>
>
```

and the use it for member `host_data_`.
```
std::vector<U, Allocator> host_data_;
```